### ISSUE TITLE
sqm2tables.py - All fixes in one place :-)

### ISSUE THREAD

**Issue by biobrad** [jtamames/SqueezeMeta#416]
Not an issue - i had a couple of problems and found this data in separate places. Adding here for the next person who needs it.

If you have run SqueezeMeta on a HPC and are now working with the results locally...

Fix SqueezeMeta sqm2tables.py issue:

In the file in the analysis folder - SqueezeMeta_conf.pl - Change path to results folder and db to match your install and db locations
`$installpath = "/home/harbj019/miniconda3/envs/newsqueezemeta/SqueezeMeta";`
`$databasepath = "/home/harbj019/dbs/db";`

If you have upgraded your SqueezeMeta Install to a later version (after 1.2) and want to work with 
data from a version 1.2 or earlier:

Fix SqueezeMeta sqm2tables.py issue:

In the file in the analysis folder - SqueezeMeta_conf.pl -

below line 55 add:
`$mappingstat = "$resultpath/10.$projectname.mappingstat";`

Run step 9 again (perl 09.summarycontigs3.pl /path/to/your/project). (use full path)
Remove the /path/to/your/project/results/tables, if present.

Save and run sqm2tables again.
`sqm2tables.py /<project> /<projectname>/table`

**Comment by fpusan** [comment]
Great, thanks for compiling this! :)

**Comment by biobrad** [comment]
Thank you for squeezemeta! You have done a lot for the research community!

**Comment by fpusan** [comment]
If you are trying to run `SQMtools` (`loadSQM` in particular) in Windows without having run `sqm2tables.py` in your Linux computer, the command will fail complaining about `sqm2tables.py`.

If this is the case, make sure you run `sqm2tables.py` in your Linux environment before migrating the data to Windows, with
`sqm2tables.py /path/to/project/ /path/to/project/results/tables`

Find more details in
https://github.com/jtamames/SqueezeMeta/wiki/Using-SQMtools-in-a-Windows-environment

**Comment by fpusan** [comment]
... and note that you also have `sqm2zip.py` to easily transfer your project to a different machine.
Also installing SQMtools has become much easier in Windows and Mac (see ReadMe!)

**Comment by srz11d** [comment]
Hello,
I am having issues with loading my files too.

I have my files in 10 separate folders (10 samples), every folder contains the following files:
 
<img width="708" alt="Screenshot 2023-09-06 at 15 57 34" src="https://github.com/jtamames/SqueezeMeta/assets/135147161/b9701310-c853-4513-82be-47b165fc5b9a">

I tried to load the files in RStudio but I seemed to fail:
I read other comments and I tried:
```
library('SQMtools')
Indoor1 = loadSQM('/Users/user/Documents/SQM_Ponds1')
```
But then I got:

```
Error in loadSQM("/Users/user/Documents/SQM_Ponds1") : 
  Directory "/Users/user/Documents/SQM_Ponds1" does not seem to contain a valid SqueezeMeta project
```
Are there any missing files? 

Thanks!

**Comment by fpusan** [comment]
did you rename your SqueezeMeta_conf.pl files?
I see one of them is names SqueezeMeta_conf_Indoor1.pl

**Comment by srz11d** [comment]
I did change the name to identify the samples, but after reading other comments, I decided to load every single folder. The folder _results_ contains a folder named _tables_. I haven't changed any tables or files, the format looks like this (for each file):
<img width="708" alt="Screenshot 2023-09-07 at 12 01 59" src="https://github.com/jtamames/SqueezeMeta/assets/135147161/d8a87160-7cba-4bfa-a9a1-8aa4063c7add">


I ran:
Outdoor2 = loadSQM('/Volumes/Intenso/Hazda/Outdoor.2')

and I got:

> Outdoor2 = loadSQM('/Volumes/Intenso/Hazda/Outdoor.2')
Loading total reads
Loading orfs
    table...
|--------------------------------------------------|
|==================================================|
    abundances...
    sequences...
    taxonomy...
Loading contigs
    table...
    abundances...
Error in dimnames(x) <- dn : 
  length of 'dimnames' [1] not equal to array extent
In addition: Warning message:
In loadSQM("/Volumes/Intenso/Hazda/Outdoor.2") :
  Your project was created with SqueezeMeta v1.5, while this is SQMtools v1.6.2. You can ignore this message if things are working fine for you, but if you experience any issue consider using the right version of SQMtools for this project


Could the version of SQM be the reason my files are not processed?

**Comment by fpusan** [comment]
yes, you should be using the SQMtools version that was shipped with SqueezeMeta v1.5

**Comment by fpusan** [comment]
This should no longer be a problem, as `sqm2tables.py` is now run as part of the main pipeline.
